packages/calypso-e2e/src/rest-api-client.ts

```diff
@@ -180,6 +202,81 @@ export class RestAPIClient {
 		return await this.sendRequest( this.getRequestURL( '1.1', '/me/sites' ), params );
 	}
 
+	/**
+	 * Deletes a site.
+	 *
+	 * If the target site has an upgrade purchased within the
+	 * sandboxed environment, it can be deleted without first
+	 * cancelling the subscription.
+	 *
+	 * Otherwise the active subscription must be first cancelled
+	 * otherwise the REST API will throw a HTTP 403 status.
+	 *
+	 * @param {SiteClosureDetails} expectedSiteDetails Expected details for the site to be deleted.
+	 * @returns {SiteDeletionResponse | null} Null if deletion was unsuccessful or not performed. SiteDeletionResponse otherwise.
+	 */
+	async deleteSite(
+		expectedSiteDetails: SiteClosureDetails
+	): Promise< SiteDeletionResponse | null > {
```

Comment:
I've chosen to define this function to return multiple types, but I am not sure how I feel about this.

On one hand, if the function needs to signify that no action was performed (for any reason) it is a trivial matter to return the `null` value. The calling function can check the return value with a simple `if ( response )` and that's it.

On the other hand, having multiple return types add to the mental workload. Additionally, it requires the calling method to first perform a check.

An alternative proposal is for the`deleteSite` method to always return a `SiteDeletionResponse`, even for invalid or no-ops. 

The interface for `SiteDeletionResponse` is defined as:

```
interface SiteDeletionResponse {
	ID: number;
	name: string;
	status: string;
}
```

By setting the value for `status` attribute to something custom like "noop" the calling method can simply check the status.